core/src/banking_stage/transaction_scheduler/thread_aware_account_locks.rs

```diff
@@ -430,6 +430,22 @@ impl ThreadSet {
     }
 }
 
+struct ThreadSetIterator(u64);
+
+impl Iterator for ThreadSetIterator {
+    type Item = ThreadId;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.0 == 0 {
+            None
+        } else {
+            let thread_id = self.0.trailing_zeros() as ThreadId;
+            self.0 &= self.0 - 1;
```

Comment:
can you add a comment in master for the sake of those not versed in the art of bit twiddling?

Reply:
https://github.com/anza-xyz/agave/pull/4006